### PR TITLE
home page function finished

### PR DESCRIPTION
**Pull request by khkyler**

### CODE REVIEW

File: src/client/app/core/api/routes.service.js [hunk 1]
```diff
@@ -7,31 +7,46 @@
   routes.$inject = ['$http', 'user'];
 
   function routes ($http, user) {
-    var _this = this;
-    _this.routes = [];
+    var vm = this;
+    vm = {
```
Comment (jbobron) [inline src/client/app/core/api/routes.service.js]: call this a service please

File: src/client/app/core/api/routes.service.js [hunk 2]
```diff
@@ -7,31 +7,46 @@
...
+      routes: [],
+      addRoute: addRoute,
+      updateRoute: updateRoute,
+      fetch: fetch
+    };
+
+    return vm;
```
Comment (jbobron) [inline src/client/app/core/api/routes.service.js]: return service here

File: src/client/app/core/api/routes.service.js [hunk 3]
```diff
@@ -7,31 +7,46 @@
...
+      fetch: fetch
+    };
+
+    return vm; 
+
+    function deleteRoute(){}
 
-    this.deleteRoute = function (){};
+    function addRoute(body, username){
+      console.log(body);
```
Comment (jbobron) [inline src/client/app/core/api/routes.service.js]: get rid of these console logs

File: src/client/app/core/api/routes.service.js [hunk 4]
```diff
@@ -7,31 +7,46 @@
...
+      console.log(vm);
+      var route = {
+        username: username,
+        route: body.route,
+        methods: {
```
Comment (jbobron) [inline src/client/app/core/api/routes.service.js]: mark as a TODO to add meaningful data later

File: src/client/app/core/api/routes.service.js [hunk 5]
```diff
@@ -7,31 +7,46 @@
...
 
-    this.addRoute = function(body){
+        },
+        headers: '',
+        body: {}
+      };
+      console.log('routes',vm.routes);
```
Comment (jbobron) [inline src/client/app/core/api/routes.service.js]: delete here and all other instances

File: src/client/app/core/api/routes.service.js [hunk 6]
```diff
@@ -7,31 +7,46 @@
...
+      vm.routes.push(body.route);
+      console.log('routes',vm.routes);
       return $http({
         method: 'POST',
-        url: '/api/users/' + user.username + '/endpoints',
+        url: '/api/users/' + username + '/endpoints',
         data: route
-      }).then(function(result) {
+      }).success(function(result) {
         console.log('ADD ROUTE SUCCESS:', result);
```
Comment (jbobron) [inline src/client/app/core/api/routes.service.js]: please add explicit `return` statement to trigger the chaining `.then` statements.

File: src/client/app/modules/home/home.controller.js [hunk 7]
```diff
@@ -20,44 +20,72 @@
...
   function HomeController($http, routes, user, checklist) {
-    var test;
-    this.formInfo = {};
-    this.formInfo.verbs = [];
-    this.routes = routes.routes;
-    this.verbs = ['GET', 'POST', 'PUT', 'DELETE', 'OPTIONS'];
+    var vm = this;
+    vm.formInfo = {};
+    vm.formInfo.methods = [];
+    vm.formInfo.body = {};
```
Comment (jbobron) [inline src/client/app/modules/home/home.controller.js]: can we change this to:
```
vm.formInfo = {
  methods = [],
  body = {}
}
```

File: src/client/app/modules/home/home.controller.js [hunk 8]
```diff
@@ -20,44 +20,72 @@
...
+    vm.routes = routes.routes;
```
Comment (jbobron) [inline src/client/app/modules/home/home.controller.js]: this is async in nature, please verify that routes.routes is meaningful at time of assignment.

File: src/client/app/modules/home/home.controller.js [hunk 9]
```diff
@@ -20,44 +20,72 @@
...
+    vm.user = {};
```
Comment (jbobron) [inline src/client/app/modules/home/home.controller.js]: do we need to make a duplicate of user or can we keep the user on the user factory?

File: src/client/app/modules/home/home.controller.js [hunk 10]
```diff
@@ -20,44 +20,72 @@
...
-    * Activate gets called on module load and calls fetch which fetches user data
-    * from the database and displays it for the user
+    * Process to capture and add route information from the user then store them in
+    * the database
     */
-    this.activate = function() {
-      user.registerCb(function(){
-        test = this.getUser();
+    function addRoute() {
+      routes.addRoute(vm.formInfo, vm.user.username).then(function() {
```
Comment (jbobron) [inline src/client/app/modules/home/home.controller.js]: please put `.then` on the next line and indent.

File: src/client/app/modules/home/home.controller.js [hunk 11]
```diff
@@ -20,44 +20,72 @@
...
+        vm.formInfo.route = '';
```
Comment (jbobron) [inline src/client/app/modules/home/home.controller.js]: I think this is to clear the form? if so, please add it in a comment.

File: src/client/app/modules/home/home.controller.js [hunk 12]
```diff
@@ -20,44 +20,72 @@
...
-    * the database
+    * Activate gets called on module load and calls fetch which fetches user data
+    * from the database and displays it for the user
     */
-    this.addRoute = function() {
-      console.log(this.formInfo);
-      routes.addRoute(this.formInfo);
-      this.formInfo.route = '';
-    };
+    function activate() {
```
Comment (jbobron) [inline src/client/app/modules/home/home.controller.js]: take `activate` and move it to line 40ish

**Comment by jbobron** [comment]
Review finished. please see comments inline

File: src/client/app/modules/home/home.controller.js (hunk 8 above)
Comment (khkyler) [inline src/client/app/modules/home/home.controller.js]: This is required to link the two arrays together for two-way data binding

**Pull request merged**